container/sagemaker/proxy_client.py

```diff
@@ -64,13 +66,16 @@ def add_model(self, model_name, base_path, model_platform='tensorflow'):
                 req = model_management_pb2.ReloadConfigRequest()
                 req.config.CopyFrom(model_server_config)
 
-                self.stub.HandleReloadConfigRequest(req)
+                self.stub.HandleReloadConfigRequest(request=req,
+                                                    timeout=GRPC_REQUEST_TIMEOUT_IN_SECONDS,
+                                                    wait_for_ready=True)
                 self._add_model_to_config_file(model_name, base_path, model_platform)
             except grpc.RpcError as e:
                 if e.code() is grpc.StatusCode.INVALID_ARGUMENT:
-                    raise Exception(409, e.details())
-                else:
-                    raise Exception(e.code(), e.details())
+                    raise requests.exceptions.RequestException(409, e.details())
```

Comment:
why change exception type? and why return a `requests` exception when we aren't even using that package?

Reply:
looking at caller below... if you need a structured exception with an error number and a message, create a custom exception class instead of borrowing one from some other package.

Reply:
Created a MultiModelException.